j/k scrolling and spacebar toggeling for ccmake

pseyfert opened the pull request:
With gmail and facebook supporting j/k scrolling, I was surprised not to find it in ccmake.

bradking commented:
@mathstuf

mathstuf commented on Source/CursesDialog/cmCursesMainForm.cxx:
```diff
@@ -854,7 +854,7 @@ void cmCursesMainForm::HandleInput()
       // and the label field for the next entry is index+1
       // (index always corresponds to the value field)
-      else if (key == KEY_UP || key == ctrl('p')) {
+      else if ( key == KEY_UP || key == ctrl('p') || key == 'k' ) {
```
Please do not add spaces inside of parentheses.

mathstuf commented on Source/CursesDialog/cmCursesStringWidget.cxx:
```diff
@@ -96,16 +96,25 @@ bool cmCursesStringWidget::HandleInput(int& key, cmCursesMainForm* fm,
 
-    // If resize occurred during edit, move out of edit mode
-    if (!this->InEdit && (key != 10 && key != KEY_ENTER)) {
+    // If resize occured during edit, move out of edit mode
```
`occurred` is spelled properly.

mathstuf commented:
I like the idea. Documentation improvements would be nice.

pseyfert commented:
do you mean in-code comments or the user manual?

mathstuf commented:
I meant user manual, but it seems we don't document the existing keybindings. I've opened [an issue](https://gitlab.kitware.com/cmake/cmake/issues/16203) for it.

pseyfert commented:
added my key bindings to the help in ccmake now.

Along, I did some minor fixes to the help:

2791831 was just wrong
858cff9 'd' isn't actually active in insert mode (there it's just the letter 'd') so moved it to the commands section

and then, not in the help but in the command listing at the bottom of the screen, I added the explanation that 'd' is available to delete an entry (c51f801).

bradking commented:
Thanks.  Please squash the commits down into the logically distinct changes (or possibly just one commit).

pseyfert commented:
okay, squashed it into these three:
* the VIM binding related changes
* update of the [d] documentation
* a typo

mathstuf commented:
LGTM overall; haven't tested it personally though.

bradking commented:
Thanks.  I've applied the changes as commits 31b6cf41c5a4e17ea97e0c94f0e81b9a18ce96b8, 9a1b6c6037869edba1c44dcfa06da8201653ce1b, and e34e9c2705c9e0bb568f5f69a655a9d425ef8adb.
